### ISSUE TITLE
Mounting /dev/pts fails if no /proc

### ISSUE THREAD

**Issue by alexlarsson** [projectatomic/bubblewrap#13]
This command fails:
```
$ bwrap --unshare-pid --ro-bind / / --dev /dev  sh
Can't mount devpts on /newroot/dev/pts: Invalid argument
```
While this works:
```
$ bwrap --unshare-pid --ro-bind / / --dev /dev --proc /proc sh
sh-4.3$ 
```
No idea why...

**Comment by runcom** [comment]
it's not working for me at all:
```
$ ./bwrap --unshare-pid --ro-bind / / --dev /dev --proc /proc sh
Can't mount devpts on /newroot/dev/pts: Invalid argument
```
also, just using `--dev /dev` triggers this error for me (no need for `--unshare-pid` in the first command above.

`strace`'ing doesn't help that much also.
with pid unsharing:
```
...
readlink("/proc/self/fd/1", "/dev/pts/1", 4095) = 10
stat("/dev/pts/1", {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
getuid()                                = 27212
getgid()                                = 27212
open("/proc", O_RDONLY|O_PATH)          = 3
mkdir("/run/user/27212/.bubblewrap", 0755) = -1 EEXIST (File exists)
eventfd2(0, EFD_CLOEXEC|EFD_NONBLOCK)   = 4
rt_sigprocmask(SIG_BLOCK, [CHLD], NULL, 8) = 0
clone(child_stack=0, flags=CLONE_NEWNS|CLONE_NEWUSER|CLONE_NEWPID|SIGCHLD) = 29837
openat(3, "self/fd", O_RDONLY|O_DIRECTORY|O_PATH) = 5
fstat(5, {st_mode=S_IFDIR|0500, st_size=0, ...}) = 0
fcntl(5, F_GETFL)                       = 0x210000 (flags O_RDONLY|O_DIRECTORY|O_PATH)
fcntl(5, F_SETFD, FD_CLOEXEC)           = 0
getdents(5, 0x1f1f1b0, 32768)           = -1 EBADF (Bad file descriptor)
close(5)                                = 0
signalfd4(-1, [CHLD], 8, SFD_CLOEXEC|SFD_NONBLOCK) = 5
poll([{fd=5, events=POLLIN}, {fd=4, events=POLLIN}], 2, -1Can't mount devpts on /newroot/dev/pts: Invalid argument
) = 1 ([{fd=5, revents=POLLIN}])
read(4, 0x7ffe3dfe5f38, 8)              = -1 EAGAIN (Resource temporarily unavailable)
read(5, "\21\0\0\0\0\0\0\0\1\0\0\0\215t\0\0Lj\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 128) = 128
exit_group(1)                           = ?
+++ exited with 1 +++
```
w/o pid unsharing:
```
readlink("/proc/self/fd/1", "/dev/pts/1", 4095) = 10
stat("/dev/pts/1", {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
getuid()                                = 27212
getgid()                                = 27212
open("/proc", O_RDONLY|O_PATH)          = 3
mkdir("/run/user/27212/.bubblewrap", 0755) = -1 EEXIST (File exists)
rt_sigprocmask(SIG_BLOCK, [CHLD], NULL, 8) = 0
clone(child_stack=0, flags=CLONE_NEWNS|CLONE_NEWUSER|SIGCHLD) = 29904
openat(3, "self/fd", O_RDONLY|O_DIRECTORY|O_PATH) = 4
fstat(4, {st_mode=S_IFDIR|0500, st_size=0, ...}) = 0
fcntl(4, F_GETFL)                       = 0x210000 (flags O_RDONLY|O_DIRECTORY|O_PATH)
fcntl(4, F_SETFD, FD_CLOEXEC)           = 0
getdents(4, 0x1fe11b0, 32768)           = -1 EBADF (Bad file descriptor)
close(4)                                = 0
signalfd4(-1, [CHLD], 8, SFD_CLOEXEC|SFD_NONBLOCK) = 4
poll([{fd=4, events=POLLIN}], 1, -1Can't mount devpts on /newroot/dev/pts: Invalid argument
)    = 1 ([{fd=4, revents=POLLIN}])
read(4, "\21\0\0\0\0\0\0\0\1\0\0\0\320t\0\0Lj\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 128) = 128
exit_group(1)                           = ?
+++ exited with 1 +++
```
stracing just `./bwrap --ro-bind / / --dev /dev sh` instead gives:
```
readlink("/proc/self/fd/1", "/dev/pts/1", 4095) = 10
stat("/dev/pts/1", {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
getuid()                                = 27212
getgid()                                = 27212
open("/proc", O_RDONLY|O_PATH)          = 3
mkdir("/run/user/27212/.bubblewrap", 0755) = -1 EEXIST (File exists)
rt_sigprocmask(SIG_BLOCK, [CHLD], NULL, 8) = 0
clone(child_stack=0, flags=CLONE_NEWNS|CLONE_NEWUSER|SIGCHLD) = 29934
openat(3, "self/fd", O_RDONLY|O_DIRECTORY|O_PATH) = 4
fstat(4, {st_mode=S_IFDIR|0500, st_size=0, ...}) = 0
fcntl(4, F_GETFL)                       = 0x210000 (flags O_RDONLY|O_DIRECTORY|O_PATH)
fcntl(4, F_SETFD, FD_CLOEXEC)           = 0
getdents(4, 0x1dad180, 32768)           = -1 EBADF (Bad file descriptor)
close(4)                                = 0
signalfd4(-1, [CHLD], 8, SFD_CLOEXEC|SFD_NONBLOCK) = 4
poll([{fd=4, events=POLLIN}], 1, -1Can't mount devpts on /newroot/dev/pts: Invalid argument
)    = 1 ([{fd=4, revents=POLLIN}])
read(4, "\21\0\0\0\0\0\0\0\1\0\0\0\356t\0\0Lj\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 128) = 128
exit_group(1)                           = ?
+++ exited with 1 +++
```

**Comment by alexlarsson** [comment]
What kernel are you running this on, and how did you build bubblewrap, setuid? setcaps or nonprivileged?

**Comment by alexlarsson** [comment]
Also, for stracing, you need to use "-f".

**Comment by mrunalp** [comment]
Interestingly, this seems to work in the latest kernel.

```
[mrunal@dhcp-16-129 bubblewrap]$ bwrap --unshare-pid --ro-bind / / --dev /dev  sh
sh-4.3$ 
sh-4.3$ exit
exit
[mrunal@dhcp-16-129 bubblewrap]$ uname -a
Linux dhcp-16-129.sjc.redhat.com 4.6.0-0.rc4.git3.1.fc25.x86_64 #1 SMP Fri Apr 22 16:16:10 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by mrunalp** [comment]
Never mind. I see that @alexlarsson fixed this in #48

**Comment by alexlarsson** [comment]
Yeah, closing this.
